src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -47,20 +73,38 @@ public async Task Invoke(HttpContext context)
         }
 
         var endpoint = context.GetEndpoint();
-
         if (endpoint != null)
         {
             // EndpointRoutingMiddleware uses this flag to check if the Authorization middleware processed auth metadata on the endpoint.
             // The Authorization middleware can only make this claim if it observes an actual endpoint.
             context.Items[AuthorizationMiddlewareInvokedWithEndpointKey] = AuthorizationMiddlewareWithEndpointInvokedValue;
         }
 
-        // IMPORTANT: Changes to authorization logic should be mirrored in MVC's AuthorizeFilter
-        var authorizeData = endpoint?.Metadata.GetOrderedMetadata<IAuthorizeData>() ?? Array.Empty<IAuthorizeData>();
+        // Use the computed policy for this endpoint if we can
+        AuthorizationPolicy? policy = null;
+
+        var canCachePolicy = _policyProvider.CanCachePolicy && _policyCache != null && endpoint != null;
```

Comment:
Can can compute most of this in the constructor right?

Reply:
Yup, done